test/Microsoft.AspNet.Mvc.TagHelpers.Test/EnvironmentTagHelperTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Test
+{
+    public class EnvironmentTagHelperTest
+    {
+        [Theory]
+        [InlineData("Development")]
+        [InlineData("development")]
+        [InlineData("DEVELOPMENT")]
+        [InlineData(" development")]
+        [InlineData("development ")]
+        [InlineData(" development ")]
+        [InlineData("Development,Production")]
+        [InlineData("Production,Development")]
+        [InlineData("Development , Production")]
+        [InlineData(" Development,Production ")]
+        [InlineData("Development,Staging,Production")]
+        [InlineData("Staging,Development,Production")]
+        [InlineData("Staging,Production,Development")]
+        public void ShowsContentWhenCurrentEnvironmentIsSpecified(string namesAttribute)
+        {
+            ShouldShowContent(namesAttribute);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        [InlineData("  ")]
+        [InlineData(", ")]
+        [InlineData("   , ")]
+        [InlineData(",")]
+        [InlineData(",,")]
+        [InlineData(",,,")]
+        [InlineData(",,, ")]
+        public void ShowsContentWhenNoEnvironmentIsSpecified(string namesAttribute)
+        {
+            ShouldShowContent(namesAttribute);
+        }
+
+        [Theory]
+        [InlineData("NotDevelopment")]
+        [InlineData("NOTDEVELOPMENT")]
+        [InlineData("NotDevelopment,AlsoNotDevelopment")]
+        [InlineData("Doesn'tMatchAtAll")]
+        [InlineData("Development and a space")]
+        [InlineData("Development and a space,SomethingElse")]
+        public void DoesNotShowContentWhenCurrentEnvironmentIsNotSpecified(string namesAttribute)
+        {
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object> { { "names", namesAttribute } },
+                content: content);
+            var output = MakeTagHelperOutput("environment");
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = "Development";
```

Comment:
would like to see a test showing a match and a non-match when `EnvironmentName` is something else.  current tests would all pass if the tag helper hard-coded "Development".